### DISCUSSION TITLE
HMR (hot module reloading), a la Vite?

### DISCUSSION THREAD

**Discussion by taviks** [lumeland/lume#336] [Ideas]
Love this project! 

Any chance some type of HMR support is (or will be) possible?

Coming from Vite-based projects loosing page state on reloads feels super no bueno.

**Comment by oscarotero** [comment]
HMR is hard to implement and probably out of the scope of Lume.
It depends on the frontend framework (react, vue, etc) and the bundle library. The library used by Lume is esbuild that doesn't support HMR https://github.com/evanw/esbuild/issues/464

**Reply by taviks** [reply]
I do wonder if it'd be possible to use something like Vite only in development (https://dev.to/vonheikemen/vite-and-deno-an-experiment-1jdf).

**Reply by oscarotero** [reply]
Probably it's possible to combine Lume and Vite (for example, serve client stuff from Vite server). But take into account that Lume is a static site generator (to create the whole site) and Vite is a frontend development tool (to work on the frontend), so Lume never will do what Vite does. But at said, perhaps a Vite plugin (I don't know if Vite is stable enough in Deno) could be possible.

**Reply by simplerethink** [reply]
Is HMR and domdiffing the same thing? [11ty 2.0 is coming out with domdiff](https://www.11ty.dev/docs/dev-server/). Maybe something to look at without going full Vite?

**Reply by oscarotero** [reply]
Not sure if this would work with some frontend frameworks like React or Vue, that store internally a virtual dom with the state of the app.
